xbmc/pvr/timers/PVRTimers.cpp

```diff
@@ -796,11 +796,21 @@ CPVRTimerInfoTagPtr CPVRTimers::GetRecordingTimerForRecording(const CPVRRecordin
       if (timersEntry->IsRecording() &&
           !timersEntry->IsTimerRule() &&
           timersEntry->m_iClientId == recording.ClientID() &&
-          timersEntry->m_iClientChannelUid == recording.ChannelUid() &&
-          timersEntry->StartAsUTC() <= recording.RecordingTimeAsUTC() &&
-          timersEntry->EndAsUTC() >= recording.EndTimeAsUTC())
+          timersEntry->m_iClientChannelUid == recording.ChannelUid())
       {
-        return timersEntry;
+        // match epg event uids, if available
+        if (timersEntry->UniqueBroadcastID() == recording.BroadcastUid())
+        {
+          if (timersEntry->UniqueBroadcastID() != EPG_TAG_INVALID_UID)
```

Comment:
These two if statements could be combined

Reply:
Yep. I will change that.